src/pk-engine.c

```diff
@@ -1755,6 +1755,38 @@ pk_engine_offline_method_call (GDBusConnection *connection_, const gchar *sender
 		g_dbus_method_invocation_return_value (invocation, value);
 		return;
 	}
+	if (g_strcmp0 (method_name, "GetResults") == 0) {
+		g_autoptr(PkResults) results = NULL;
+		gboolean success = FALSE;
+		g_autoptr(PkError) pk_error = NULL;
+		const gchar *err_descr = "";
+		GVariant *value = NULL;
+
+		results = pk_offline_get_results (&error);
+		if (results == NULL || pk_results_get_exit_code (results) != PK_EXIT_ENUM_SUCCESS) {
```

Comment:
```suggestion
		if (results == NULL) {
```
This should probably only throw an error if there was one returned...